packages/okta-react/src/Auth.js

```diff
@@ -164,21 +185,44 @@ export default class Auth {
     return this._oktaAuth.token.getWithRedirect(params);
   }
 
-  setFromUri (fromUri) {
+  setFromUri(fromUri) {
     // Use current history location if fromUri was not passed
     const referrerPath = fromUri
       ? { pathname: fromUri }
       : this._history.location;
-    localStorage.setItem(
-      'secureRouterReferrerPath',
-      JSON.stringify(referrerPath)
-      );
+    localStorage.setItem( 'secureRouterReferrerPath', JSON.stringify(referrerPath) );
   }
 
-  getFromUri () {
+  getFromUri() {
     const referrerKey = 'secureRouterReferrerPath';
     const location = JSON.parse(localStorage.getItem(referrerKey) || '{ "pathname": "/" }');
     localStorage.removeItem(referrerKey);
     return location;
   }
+
+  on( event, callback ) {
+    // FIXME: restricted events?
+    const eventId = this.eventCount++;
+    this._listeners[event] = this._listeners[event] || {};
+    this._listeners[event][eventId] = callback;
+    return () => { 
```

Comment:
If devs eventually start to use this function, return the unregister function may looks like a magic. Provide an `off` function can provide better readability.

Reply:
This is a fairly common pattern, and it shelters us from exposing the internal `eventId` (which means we can later change the eventId mechanism).  

I'll do a little research into alternatives.

Reply:
I understand this pattern definitely work. Also if it's only private method for SDK internally, how about add _ to mark it as private.

Reply:
The method is public, though as a return value it is without a name, so I'm not sure I follow.

If you mean making the eventId intoi `_eventId`, that's a good call.

Reply:
My question was about the access modifier of `on` method.

Reply:
`on()` is a public method, albeit one that would be used rarely, since most users should be fine with the provided `<Security>` component.